packages/models/src/resource-binding/manager.ts

```diff
@@ -28,43 +36,72 @@ export class Manager extends Actor<object, MessagePayload<ResourceBindingManager
   }
 
   private update(pollingInterval = 1000) {
+    let semaphore = 0
```

Comment:
Maybe it can define as a `boolean` type?

Reply:
> Maybe it can define as a `boolean` type?

The `semaphore` might be larger than 1, `update` will be processed only when the `semaphore` is equal to 1.

Reply:
How about like this?
```
let semaphore = false 
return setInterval(async () => {
      if (!semaphore) {
        semaphore = true
        // do something
        semaphore = false
      }
    }, pollingInterval)
```

Reply:
Suppose there are two processes `A` and `B`.
When `A` read the `semaphore`, it is `false`. At the same time, `B` read the `semaphore`, and the semaphore has not  been set to `true` yet by `A`, it is still `false`. Because `if (!semaphore) {` and `semaphore = true` are not atomic.
I think it is possible.

Reply:
Got it, but the semaphore is not shared for two processes. And for one process it will execute in sequence.